modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/application/ApplicationDescriptorFactoryTestCase.java

```diff
@@ -84,11 +84,16 @@ private static File getResourceFile(String resource) {
   public static final String JAR_FILE_NAME = "test.jar";
 
   @Rule
-  public SystemProperty remoteMavenRepository = new SystemProperty("muleRuntimeConfig.maven.repositoryLocation",
-                                                                   discoverProvider(ApplicationDescriptorFactoryTestCase.class
-                                                                       .getClassLoader()).getLocalRepositorySuppliers()
-                                                                           .environmentMavenRepositorySupplier().get()
-                                                                           .getAbsolutePath());
+  public SystemProperty repositoryLocation = new SystemProperty("muleRuntimeConfig.maven.repositoryLocation",
+                                                                discoverProvider(ApplicationDescriptorFactoryTestCase.class
+                                                                    .getClassLoader()).getLocalRepositorySuppliers()
+                                                                        .environmentMavenRepositorySupplier().get()
+                                                                        .getAbsolutePath());
+
+  @Rule
+  public SystemProperty remoteMavenRepository =
```

Comment:
If that is the problem, then it should be enough to declare a dependency on the deployment module pom to the HTTP extension. I don't think that that will introduce a circular dependency